src/services/axios-http-data.service.ts

```diff
@@ -75,53 +84,73 @@ export class AxiosHttpDataService extends AbstractHttpDataService {
    * @param axiosRequestCallFn The axios operation function
    */
   private async axiosHttpCall<K>(
     url: string,
     queryParams: IQueryParams,
     headers: IHeaders,
+    timeout: number,
+    retries: number,
     axiosRequestCallFn: (fnUrl: string, fnRequestConfig: AxiosRequestConfig) => Promise<AxiosResponse<K>>
   ): Promise<IApiResponse<K>> {
 
     const requestConfig: AxiosRequestConfig = {
       headers,
-      params: queryParams
+      params: queryParams,
+      timeout
     };
 
-    try {
+    for (let attempt = 0; attempt <= retries; attempt++) {
+      try {
+
+        const response = await axiosRequestCallFn(url, requestConfig);
+
+        const apiResponse: IApiResponse<K> = {
+          body: response.data,
+          status: response.status,
+          headers: response.headers as IHeaders
+        };
 
-      const response = await axiosRequestCallFn(url, requestConfig);
+        return apiResponse;
 
-      const apiResponse: IApiResponse<K> = {
-        body: response.data,
-        status: response.status,
-        headers: response.headers as IHeaders
-      };
+      } catch (error) {
 
-      return apiResponse;
+        const e: AxiosError = error as AxiosError;
 
-    } catch (error) {
+        const timeoutRegExp = /^timeout of [0-9]+ms exceeded$/;
 
-      const e: AxiosError = error as AxiosError;
+        if (attempt < retries && timeoutRegExp.test(e.message)) {
+          continue;
+        }
 
-      const errorData = {
-        name: e.name,
-        message: e.message,
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-        data: e.response?.data || `API Call Failed. ${e.message}`
-      };
+        const errorData = {
+          name: e.name,
+          message: e.message,
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+          data: e.response?.data || `API Call Failed. ${e.message}`
+        };
 
-      const apiResponse: IApiResponse<unknown> = {
-        body: e.response?.data || {},
-        error: errorData,
-        status: e.response?.status || StatusCodes.INTERNAL_SERVER_ERROR,
-        headers: e.response?.headers as IHeaders
-      };
+        const apiResponse: IApiResponse<unknown> = {
+          body: e.response?.data || {},
+          error: errorData,
+          status: e.response?.status || StatusCodes.INTERNAL_SERVER_ERROR,
+          headers: e.response?.headers as IHeaders
+        };
 
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      return apiResponse as IApiResponse<any>;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        return apiResponse as IApiResponse<any>;
+      }
     }
-  }
 
+    const fallbackApiResponse: IApiResponse<unknown> = {
+      body: {},
+      status: StatusCodes.INTERNAL_SERVER_ERROR
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return fallbackApiResponse as IApiResponse<any>;
```

Comment:
Just thinking more on this...
The underlying error here is that the http calls are timing out.
We should capture this as a `lastReportedErrorResponse` and return that instead rather than an error that contains no information back to the calling application code